x/evm/types/command.go

```diff
@@ -138,6 +138,25 @@ func NewApproveContractCallCommand(
 	}
 }
 
+// NewApproveContractCallCommandFromGeneralMessage creates a command to approve contract call
+func NewApproveContractCallCommandFromGeneralMessage(
```

Comment:
do we have approve contract calls for anything but GMP?

Reply:
EVM to EVM doesn't create a general message at least, so uses a different function. I did refactor this to make it more generic though.